rust/src/lib/unit_tests/route_rule.rs

```diff
@@ -320,3 +320,27 @@ fn test_route_rule_auto_priority_increasing_from_empty() {
     );
     assert_eq!(rules[2].priority, Some(30002));
 }
+
+#[test]
+fn test_route_rule_sorting() {
+    let mut rules = vec![
+        RouteRuleEntry {
+            priority: Some(200),
+            ..Default::default()
+        },
+        RouteRuleEntry {
+            priority: Some(100),
+            ..Default::default()
+        },
+        RouteRuleEntry {
+            priority: Some(300),
+            ..Default::default()
+        },
+    ];
+
+    rules.sort_unstable();
```

Comment:
What is the purpose of this unit test? The test does not call any internal function, I think that what is missing in this PR is to have a integration test which has the state with several route rule entries, then the test applies the desired state and retrieves the state with `libnmstate.show()`  to see if the route rules are properly sorted.